File: packages/motion_sleep/ios/Classes/Repository/MotionSleepInterface.swift
//
//  MotionSleepInterface.swift
//  motion_sleep
//
//  Created by Michael Jajou on 2/23/23.
//

import Foundation

protocol MotionSleepInterface {
    func isActivityAvailable() -> Bool
    func requestAuthorization()
    func fetchMostRecentSleepSession(
        start: Date,
        end: Date,
        sleepTime: SleepTime,
        completion: @escaping (Result<SleepSession, Error>) -> Void
    )
    func fetchSleepSessions(
        start: Date,
        end: Date,
        sleepTime: SleepTime,
        completion: @escaping (Result<[SleepSession], Error>) -> Void
    )
    func fetchActivities(
        start: Date,
        end: Date,
        completion: @escaping (Result<[MotionActivity], Error>) -> Void
    )
}

File: packages/motion_sleep/lib/src/motion_sleep_interface.dart
part of motion_sleep;

abstract class MotionSleepInterface {
  Future<MotionAuthorizationStatus> fetchAuthorizationStatus();
  Future<bool> isActivityAvailable();
  Future<void> requestAuthorization();
  Future<SleepSession?> fetchMostRecentSleepSession({
    required DateTime start,
    required DateTime end,
    required SleepTime sleepTime,
  });
  Future<List<SleepSession>> fetchSleepSessions({
    required DateTime start,
    required DateTime end,
    required SleepTime sleepTime,
  });
  Future<List<MotionActivity>> fetchActivities({
    required DateTime start,
    required DateTime end,
  });
}
